flink-runtime/src/main/java/org/apache/flink/runtime/state/DefaultKeyedStateStore.java

```diff
@@ -93,6 +95,18 @@ public DefaultKeyedStateStore(KeyedStateBackend<?> keyedStateBackend, ExecutionC
 		}
 	}
 
+	@Override
+	public <UK, UV> MapState<UK, UV> getMapState(MapStateDescriptor<UK, UV> stateProperties) {
+		requireNonNull(stateProperties, "The state properties must not be null");
+		try {
+			stateProperties.initializeSerializerUnlessSet(executionConfig);
+			MapState<UK, UV> originalState = getPartitionedState(stateProperties);
+			return new UserFacingMapState<>(originalState);
+		} catch (Exception e) {
+			throw new RuntimeException("Error while getting state", e);
```

Comment:
What I meant was using a more specific subclass of `RuntimeException` (e.g. `IllegalAccess`,`IllegalState`, `IndexOutOfBounds`, etc.) that better reflects the error, but still can leave the method signature unchanged.